src/components/radio-group-item/radio-group-item.tsx

```diff
@@ -84,10 +116,12 @@ export class RadioGroupItem {
             "label--outline": appearance === "outline"
           }}
         >
-          {this.icon && this.iconPosition === "start" ? iconEl : null}
+          {this.icon && this.iconPosition === "start" && !this.iconStart ? iconEl : null}
```

Comment:
Can you refactor this to just render `iconStart` like you have in the accordion? That way its less confusing here.

Reply:
I don't think we can render iconStart here as we did in accordion, the issue is with the mapping of the icon to `iconStart` when `iconPosition` is set to `start` and able to mutate either of `iconStart` or `iconEnd`   when `iconPosition` changes but not `icon` value.